spec/lib/cds_importer/entity_mapper/additional_code_mapper_spec.rb

```diff
@@ -1,33 +1,126 @@
 RSpec.describe CdsImporter::EntityMapper::AdditionalCodeMapper do
-  it_behaves_like 'an entity mapper', 'AdditionalCode', 'AdditionalCode' do
-    let(:xml_node) do
-      {
-        'sid' => '3084',
-        'additionalCodeCode' => '169',
-        'validityEndDate' => '1996-06-14T23:59:59',
-        'validityStartDate' => '1991-06-01T00:00:00',
-        'additionalCodeType' => {
-          'additionalCodeTypeId' => '8',
+  let(:xml_node) do
+    {
+      'sid' => '3084',
+      'additionalCodeCode' => '169',
+      'validityEndDate' => '1996-06-14T23:59:59',
+      'validityStartDate' => '1991-06-01T00:00:00',
+      'additionalCodeType' => {
+        'additionalCodeTypeId' => '8',
+      },
+      'metainfo' => {
+        'origin' => 'T',
+        'opType' => operation,
+        'transactionDate' => '2016-07-27T09:20:15',
+      },
+      'footnoteAssociationAdditionalCode' => {
+        'footnote' => {
+          'footnoteId' => '08',
+          'footnoteType' => {
+            'footnoteTypeId' => '06',
+          },
+        },
+        'metainfo' => {
+          'opType' => operation,
+          'transactionDate' => '2017-08-27T19:23:57',
+        },
+      },
+      'additionalCodeDescriptionPeriod' => {
+        'sid' => '536',
+        'additionalCodeDescription' => {
+          'description' => 'Other.',
+          'language' => {
+            'languageId' => 'EN',
+          },
+          'metainfo' => {
+            'origin' => 'T',
+            'opType' => operation,
+            'transactionDate' => '2016-07-27T09:20:14',
+          },
         },
         'metainfo' => {
           'origin' => 'T',
-          'opType' => 'U',
-          'transactionDate' => '2016-07-27T09:20:15',
+          'opType' => operation,
+          'transactionDate' => '2016-07-27T09:20:14',
         },
-      }
-    end
+      },
+    }
+  end
+
+  let(:operation) { 'C' }
 
+  it_behaves_like 'an entity mapper', 'AdditionalCode', 'AdditionalCode' do
     let(:expected_values) do
       {
         validity_start_date: Time.zone.parse('1991-06-01T00:00:00.000Z'),
         validity_end_date: Time.zone.parse('1996-06-14T23:59:59.000Z'),
         national: false,
-        operation: 'U',
+        operation: 'C',
```

Comment:
should this be:
`operation: operation`
?

Reply:
chatted offline. this is basically part of removing duplication where a new default exists on the xml node